storage/chunked_file_store.py

```diff
@@ -24,73 +25,69 @@ class ChunkedFileStore(FileStore):
                  dbDir,
                  dbName,
                  isLineNoKey: bool=False,
                  storeContentHash: bool=True,
                  chunkSize: int=1000,
                  ensureDurability: bool=True,
-                 chunkStoreConstructor=TextFileStore,
-                 defaultFile=None):
+                 chunk_creator=None,
+                 open=True):
         """
 
         :param chunkSize: number of items in one chunk. Cannot be lower then number of items in defaultFile
         :param chunkStoreConstructor: constructor of store for single chunk
         """
 
-        assert chunkStoreConstructor is not None
-
         super().__init__(dbDir,
                          dbName,
                          isLineNoKey,
                          storeContentHash,
                          ensureDurability,
-                         defaultFile=defaultFile)
+                         open=False)
 
         self.chunkSize = chunkSize
         self.itemNum = 1  # chunk size counter
         self.dataDir = os.path.join(dbDir, dbName)  # chunk files destination
-        self.currentChunk = None  # type: FileStore
+        self.currentChunk = None  # type: KeyValueStorageFile
         self.currentChunkIndex = None  # type: int
 
-        self._chunkCreator = lambda name: \
-            chunkStoreConstructor(self.dataDir,
+        # TODO: fix chunk_creator support
+        default_chunk_creator = lambda name: \
+            TextFileStore(self.dataDir,
                                   name,
                                   isLineNoKey,
                                   storeContentHash,
                                   ensureDurability)
 
-        self._initDB(dbDir, dbName)
+        self._chunkCreator = chunk_creator or default_chunk_creator
+        if open:
+            self.open()
 
-    def _prepareFiles(self, dbDir, dbName, defaultFile):
-
-        def getFileSize(file):
-            with self._chunkCreator(file) as chunk:
-                return chunk.numKeys
+    @property
+    def is_byte(self) -> bool:
+        return self.currentChunk.is_byte
 
-        path = os.path.join(dbDir, dbName)
-        os.mkdir(path)
-        if defaultFile:
-            if self.chunkSize < getFileSize(defaultFile):
-                raise ValueError("Default file is larger than chunk size")
-            firstChunk = os.path.join(path, str(self.firstChunkIndex))
-            shutil.copy(defaultFile, firstChunk)
+    def _init_db_path(self, dbDir, dbName):
+        return os.path.join(os.path.expanduser(dbDir), dbName)
 
-    def _initDB(self, dataDir, dbName) -> None:
-        super()._initDB(dataDir, dbName)
-        path = os.path.join(dataDir, dbName)
-        if not os.path.isdir(path):
+    def _init_db_file(self):
+        if not os.path.exists(self.db_path):
+            os.makedirs(self.db_path)
+        if not os.path.isdir(self.db_path):
             raise ValueError("Transactions file {} is not directory"
```

Comment:
According to `ValueError` description, this is `Inappropriate argument value (of correct type)`. In this case this is actually it is.